pkg/apis/pingcap/v1alpha1/types.go

```diff
@@ -2024,14 +2071,21 @@ type RestoreSpec struct {
 	Env []corev1.EnvVar `json:"env,omitempty"`
 	// To is the tidb cluster that needs to restore.
 	To *TiDBAccessConfig `json:"to,omitempty"`
-	// Type is the backup type for tidb cluster.
+	// Type is the backup type for tidb cluster and only used when Mode = snapshot, such as full, db, table.
 	Type BackupType `json:"backupType,omitempty"`
+	// Mode is the restore mode. such as snapshot or pitr.
+	// +kubebuilder:default=snapshot
+	Mode RestoreMode `json:"restoreMode,omitempty"`
+	// RestoredTs is the pitr restored ts.
+	RestoredTs string `json:"restoredTs,omitempty"`
 	// TikvGCLifeTime is to specify the safe gc life time for restore.
 	// The time limit during which data is retained for each GC, in the format of Go Duration.
 	// When a GC happens, the current time minus this value is the safe point.
 	TikvGCLifeTime *string `json:"tikvGCLifeTime,omitempty"`
 	// StorageProvider configures where and how backups should be stored.
 	StorageProvider `json:",inline"`
+	// LogBackupStorageProvider configures where and how log backup should be stored.
+	LogBackupStorageProvider StorageProvider `json:"logBackupStorageProvider,omitempty"`
```

Comment:
why not reuse `StorageProvider`

Reply:
PiTR restore needs a logBackup storage path, but now other configuration information is consistent with `StorageProvider` except for storage path. It is for convenience of expansion in the feature.

Reply:
can't get it, could you please provide an example?

Reply:
ok. PiTR restore needs a snapshot backup and a log backup, they are two storage path. now they can use the same `StorageProvider`, just different path. `LogBackupStorageProvider` is for convenience of that they may be not in the same `StorageProvider` in the future.

Reply:
i can take it back to [this](https://github.com/pingcap/tidb-operator/pull/4648/commits/62fa62993361d6ce78bec14922178e75533cf066#diff-c56b69715506be0578996118127b15255d8f5ac302da026bf2f5a7ecf4ac6685R1528) if both of you think it's not a good way. @KanShiori @csuzhangxc